frontend/lib/resolution/default-functions.cpp

```diff
@@ -297,40 +297,55 @@ static void collectFields(const AstNode* ast,
   }
 }
 
-static void initHelper(Context* context,
+static const BuilderResult& buildInitializer(Context* context, ID typeID);
+
+static bool initHelper(Context* context,
                        Builder* builder,
                        const AggregateDecl* typeDecl,
                        const Location& dummyLoc,
-                       AstList& formals, AstList& superArgs, AstList& stmts,
-                       bool isChild = true) {
+                       AstList& formals, AstList& superArgs, AstList& stmts) {
+  // Return 'true' if a super.init call is necessary
+  bool addSuperInit = false;
+
+  // Check if we need a super.init() call. If the parent has a default
+  // initializer, add arguments to the super.init() and formals to the
```

Comment:
If there's no `super.init` call, aren't we supposed to generate one with zero args, rather than filling in arguments? (https://chapel-lang.org/docs/language/spec/classes.html#initializing-inherited-classes)

Reply:
That's for user-defined initializers. Default initializers are intended to support arguments for the parent/ancestor classes provided those classes don't have a user-defined initializer.